### PR TITLE
MuJoCo and IsaacGym Infra Update

### PR DESCRIPTION
**Pull request by EmptyBlueBox**
1. Align the `get_states` function.  
2. Update the `h1` reward function to ensure compatibility with both `mjcf` and `urdf`.  
3. Simplify the SB3 training code in line with infrastructure updates.

### CODE REVIEW

File: metasim/sim/isaacgym/isaacgym.py [hunk 1]
```diff
@@ -453,49 +452,51 @@ def get_states(self, env_ids: list[int] | None = None) -> list[EnvState]:
...
+                            self._body_info[obj.name]["global_indices"][body_name], 3:7
+                        ].cpu(),
+                        "vel": self._rigid_body_states[
+                            self._body_info[obj.name]["global_indices"][body_name], 7:10
+                        ].cpu(),
+                        "ang_vel": self._rigid_body_states[
+                            self._body_info[obj.name]["global_indices"][body_name], 10:
+                        ].cpu(),
+                    }
+                    env_state[object_type][obj.name][body_name] = body_state
```
Comment (Fisher-Wang) [inline metasim/sim/isaacgym/isaacgym.py:470]: Would be better to use `env_state[object_type][obj.name]['body'][body_name]` to access body link states? 'body' sounds at the same level as `pos`, `rot`, ...?

File: metasim/sim/mujoco/mujoco.py [hunk 2]
```diff
@@ -267,42 +267,87 @@ def get_states(self, env_ids: list[int] | None = None) -> list[dict]:
...
-                            self.physics.data.sensordata[sensor_id : sensor_id + 3], dtype=torch.float32
-                        )
-
-            state[f"metasim_site_{site.name.split('/')[-1]}"] = {
-                "pos": torch.tensor(self.physics.data.site_xpos[site_id], dtype=torch.float32),
-                "rot": torch.tensor(self.physics.data.site_xmat[site_id], dtype=torch.float32),
-                "vel": linear_vel if linear_vel is not None else None,
-                "ang_vel": angular_vel if angular_vel is not None else None,
-            }
+        # NOTE: Site does not compatible with urdf or usd, so we don't use it
```
Comment (Fisher-Wang) [inline metasim/sim/mujoco/mujoco.py:326]: feel free to delete them if they are no longer useful

**Comment by Fisher-Wang** [comment]
Thank you Yutong for such a nice and clean pr!

File: metasim/sim/isaacgym/isaacgym.py (hunk 1 above)
Comment (EmptyBlueBox) [reply metasim/sim/isaacgym/isaacgym.py:470]: Certainly, it looks much clearer!

File: metasim/sim/mujoco/mujoco.py (hunk 2 above)
Comment (EmptyBlueBox) [reply metasim/sim/mujoco/mujoco.py:326]: Sure!

File: metasim/sim/mujoco/mujoco.py [hunk 3]
```diff
@@ -259,50 +259,74 @@ def _get_actuator_states(self, obj_name):
...
             state[object_type][obj.name].update(self._get_joint_states(obj.name))
+            ## Actuator states -- for robot
             if obj == self._robot:
                 state[object_type][obj.name].update(self._get_actuator_states(obj.name))
 
+        ## Body states -- for both robot and object
+        object_names = [obj.name for obj in self.objects]
+        robot_names = [self._robot.name]
+        for obj in self.objects + [self._robot]:
+            state["robots"][obj.name]["body"] = {}
```
Comment (Fisher-Wang) [inline metasim/sim/mujoco/mujoco.py:276]: There should also fit the "objects" key case. Please refer to https://github.com/RoboVerseOrg/RoboVerse/pull/52/files#diff-04c010ddb6c1a6d3e9dcf94a365dad4815713a9ecaefc218056e2edfa85628b0L262-L266

**Comment by Fisher-Wang** [comment]
And please test the pr using
```
python metasim/scripts/replay_demo.py --task=close_box --sim=mujoco
```

File: metasim/sim/mujoco/mujoco.py (hunk 3 above)
Comment (EmptyBlueBox) [reply metasim/sim/mujoco/mujoco.py:278]: fixed!

**Comment by Fisher-Wang** [comment]
All test passed, you are awesome!!
